src/gui/guiFormSpecMenu.cpp

```diff
@@ -779,6 +780,58 @@ void GUIFormSpecMenu::parseImage(parserData* data, const std::string &element)
 	errorstream<< "Invalid image element(" << parts.size() << "): '" << element << "'"  << std::endl;
 }
 
+void GUIFormSpecMenu::parseAnimatedImage(parserData *data, const std::string &element)
+{
+	std::vector<std::string> parts = split(element, ';');
+
+	if (parts.size() != 3 &&
+			(parts.size() < 3 || m_formspec_version <= FORMSPEC_API_VERSION)) {
+		errorstream << "Invalid animated image element(" << parts.size()
+				<< "): '" << element << "'"  << std::endl;
+		return;
+	}
+
+	std::vector<std::string> v_pos   = split(parts[0], ',');
+	std::vector<std::string> v_geom  = split(parts[1], ',');
+	std::string name = unescape_string(parts[2]);
+
+	MY_CHECKPOS("animated_image", 0);
+	MY_CHECKGEOM("animated_image", 1);
+
+    v2s32 pos;
+    v2s32 geom;
+
+    if (data->real_coordinates) {
+        pos = getRealCoordinateBasePos(v_pos);
+        geom = getRealCoordinateGeometry(v_geom);
+    } else {
+        pos = getElementBasePos(&v_pos);
+        geom.X = stof(v_geom[0]) * (float)imgsize.X;
+        geom.Y = stof(v_geom[1]) * (float)imgsize.Y;
+    }
```

Comment:
Please use tabulators.

Reply:
Sorry for screwing that up a second time...I've taken measures to ensure that it won't happen again.